pkg/controllers/externalsecret/externalsecret_controller.go

```diff
@@ -464,6 +464,21 @@ func shouldSkipUnmanagedStore(ctx context.Context, namespace string, r *Reconcil
 		if ref.SourceRef != nil && ref.SourceRef.SecretStoreRef != nil {
 			storeList = append(storeList, *ref.SourceRef.SecretStoreRef)
 		}
+
+		// verify that generator's controllerClass matches
+		if ref.SourceRef != nil && ref.SourceRef.GeneratorRef != nil {
+			genDef, err := r.getGeneratorDefinition(ctx, namespace, ref.SourceRef)
+			if err != nil {
+				return false, err
+			}
+			skipGenerator, err := shouldSkipGenerator(r, genDef)
+			if err != nil {
+				return false, err
+			}
+			if skipGenerator {
+				return true, nil
+			}
+		}
```

Comment:
I moved the skip check here. The generator controller class check should be at the very top of the reconcile function just like the other secretStore class check.
    
Otherwise we would return an error and as a result set the status field on the es resource - which is undesirable. The controller should completely ignore the resource instead.

Reply:
Good Idea